src/SBarcodeScanner.h

```diff
@@ -0,0 +1,214 @@
+    * \brief Function for image processing
+    * \param const QVideoFrame &frame - video frame
+    */
+    void imageProcess(SBarcodeDecoder *decoder, const QImage &image, ZXing::BarcodeFormats formats);
+
+private:
+    /*!
+    * \brief Decoder instance
+    */
+    SBarcodeDecoder m_decoder;
+
+    /*!
+    * \brief Camera instance
+    */
+    QCamera *camera;
+
+    /*!
+    * \brief Pointer to a sink
+    */
+    QPointer<QVideoSink> m_videoSink;
+
+    /*!
+    * \brief Capture area
+    */
+    QRectF m_captureRect;
+
+    /*!
+    * \brief Captured string
+    */
+    QString m_captured = "";
+
+    /*!
+    * \brief Camera session instance
+    */
+    QMediaCaptureSession m_capture;
+
+    /*!
+     * \brief An instance of a thread
+     */
+    QThread workerThread;
+
+    /*!
+     * \brief A pointer of a Worker class
+     */
+    Worker *worker;
+
+    /*!
+    * \fn void setCaptured(const QString &captured)
+    * \brief Function for setting capture string
+    * \param const QString &captured - captured string
+    */
+    void setCaptured(const QString &captured);
+
+    /*!
+    * \fn void handleFrameCaptured(const QVideoFrame &frame)
+    * \brief Function for handling video frame
+    * \param const QVideoFrame &frame - video frame
+    */
+    void handleFrameCaptured(const QVideoFrame &frame);
+
+signals:
+    /*!
+     * \brief This signal emitted when camera changed
+     */
+    void cameraChanged();
+
+    /*!
+     * \brief This signal emitted for running process in a thread
+     */
+    void process(const QImage &image);
+
+    /*!
+     * \brief This signal is emitted when sink changed
+     */
+    void videoSinkChanged();
+
+    /*!
+     * \brief This signal is emitted when capture area changed
+     * \param const QRectF &captureRect - capture area
+     */
+    void captureRectChanged(const QRectF &captureRect);
+
+    /*!
+     * \brief This signal is emitted when captured string changed
+     * \param const QString &captured - captured string
+     */
+    void capturedChanged(const QString &captured);
+
+private slots:
+    /*!
+    * \fn void initCam()
+    * \brief Function for initialization of camera
+    */
+    void initCam();
+    /*!
+    * \fn void stopCam()
+    * \brief Function for stopping camera
+    */
+    void stopCam();
+
+};
+
+class Worker : public QObject
+{
+    Q_OBJECT
+private:
+    SBarcodeScanner *_scanner;
+
+public:
+    /*!
+     * \fn Worker(SBarcodeScanner *scanner)
+     * \brief Constructor.
+     * \param SBarcodeScanner *_scanner - a pointer to scanner class.
+     */
+    Worker(SBarcodeScanner *scanner) : _scanner{scanner} { ; }
+
+public slots:
+
+    /*!
+    * \fn void process(const QImage &image)
+    * \param const QImage &image - captured image
+    * \brief An interface for processing image
+    */
+    void process(const QImage &image) {
+        _scanner->imageProcess(_scanner->getDecoder(), image, SCodes::toZXingFormat(SCodes::SBarcodeFormat::Basic));
+    }
```

Comment:
I'm a bit rusty on Qt with threads but are you sure this executes on a worker thread? -  because `_scanner` is still "living" in the original thread. Also the arguments indicate that this is supposed to be a `static` function but they're unused in the definition above

Reply:
Worker object that dependent on scanner variable passed to workerThread already. So we can access to resources of the scanner object. 
worker->moveToThread(&workerThread);    

Emitting the process signal provides to run the imageProcess function in workerThread. 
emit process();  

Without emitting, the GUI thread freezes. So emitting the signal provides to run the `imageProcess` function in worker thread. Tested on Redmi Note 7.

If you have any idea to implement it in a elegant way, please do not hesitate to share w/ me.

Reply:
If it works - it works. Generally this part is OK, I was just a bit hazy if it works as intended 😅